crates/bevy_state/src/state/transitions.rs

```diff
@@ -206,3 +207,15 @@ pub(crate) fn run_transition<S: States>(
 
     let _ = world.try_run_schedule(OnTransition { exited, entered });
 }
+
+/// Logs state transitions into console.
+///
+/// This system is provided to make debugging easier for Bevy developers.
+pub fn log_transitions<S: States>(mut transitions: EventReader<StateTransitionEvent<S>>) {
+    for transition in transitions.read() {
+        info!(
+            "Transition: {:?} => {:?}",
```

Comment:
We should add the type name here somehow, using https://doc.rust-lang.org/std/any/fn.type_name.html. 

Like Pyrious pointed out on Discord,  the derived enum debugs won't provide the type name by default.

Reply:
Already on it :)